src/main/resources/assets/admin/common/js/ui/panel/DockedPanel.ts

```diff
@@ -41,6 +41,24 @@ export class DockedPanel
         return this.deck.getPanelIndex(panel);
     }
 
+    setItemVisible<T extends Panel>(panel: T, visible: boolean): void {
+        const index = this.deck.getPanelIndex(panel);
+        if (index >= 0) {
+            const currentlyVisible = this.deck.getPanelShownIndex() == index;
+            this.items[index].toggleClass('hidden', !visible);
+            if (!visible && currentlyVisible && this.items.length >= 2) {
+                // currently visible item has been hidden, so find a substitute to show
+                for (let i = 0; i < this.items.length; i++) {
+                    const item = this.items[i];
+                    if (i !== index && !item.hasClass('hidden')) {
+                        this.navigator.selectNavigationItem(i);
+                        break;
+                    }
+                }
+            }
+        }
```

Comment:
The logic here is very difficult to follow. Also, there's no point in using yet another CSS class (`"hidden"`) if we already have `hide`/`show` on every element.

So it can be simplified like this (has to be tested):

    setItemVisible<T extends Panel>(panel: T, visible: boolean): void {
        const currentIndex = this.deck.getPanelIndex(panel);
        if (currentIndex < 0) { // The panel is not found
            return;
        }

        const isCurrentlyVisible = this.deck.getPanelShownIndex() === currentIndex;
        if (isCurrentlyVisible === visible) {
            return;
        }

        this.items[currentIndex].setVisible(visible);
        if (isCurrentlyVisible) {
            const newItemToShow = this.items.find((item, index) => item.isVisible() && currentIndex !== index);
            if (newItemToShow) {
                newItemToShow.show();
            }
        }
    }

Reply:
The reason of using 'hidden' instead of show/hide is that we need to be able to do a check if item is hidden:
- If we use regular show/hide, then isVisible() will return false for every tab if the whole panel is not currently visible.
- If we use 'hidden' class, then hasClass('hidden') will return real item state indifferently of the panel visibility.

But the logic can be simplified I guess ;)